koku/api/provider/serializers.py

```diff
@@ -326,8 +331,12 @@ def create(self, validated_data):
             Provider.objects.filter(authentication=auth).filter(billing_source=bill).filter(customer=customer).count()
         )
         if unique_count != 0:
-            error = {"Error": "A Provider already exists with that Authentication and Billing Source"}
-            raise serializers.ValidationError(error)
+            message = (
+                f"Cost management does not allow duplicate accounts. "
+                f"{validated_data['name']} already exists. Edit source settings to configure a new source."
```

Comment:
Very close!
This name should be the first source that was created.  You'll have to get that name from the `Provider.objects.filter(authentication=auth).filter(billing_source=bill).filter(customer=customer)` query.

Here's an example:

Source message was successfully pushed to platform sources.  Just need to fix the name so the user will know what source they have to edit or remove.

`GET http://localhost:3000/api/v1.0/applications/?filter[source_id]=2`
```
{
    "meta": {
        "count": 1,
        "limit": 100,
        "offset": 0
    },
    "links": {
        "first": "/api/v1.0/applications/?filter%5Bsource_id%5D=2&offset=0",
        "last": "/api/v1.0/applications/?filter%5Bsource_id%5D=2&offset=0"
    },
    "data": [
        {
            "application_type_id": "2",
            "availability_status": "unavailable",
            "availability_status_error": "Cost management does not allow duplicate accounts. Test OCP Source 2 already exists. Edit source settings to configure a new source.",
            "created_at": "2020-05-27T19:54:38Z",
            "id": "2",
            "source_id": "2",
            "updated_at": "2020-05-27T19:54:39Z",
            "tenant": "10001"
        }
    ]
}
```

Reply:
In this example I already had a source created with the same cluster-id.  Name was `Test OCP Source 1`

Reply:
OK it was tricky than I thought. I couldn't execute queries inside the atomic block (which is a little odd because `ProviderBillingSource.objects.get_or_create(**billing_source)` seems like one). Anyway, I moved all that block out of the atomic block.

Reply:
Looks great!  I'm going to try to move that transaction back before the get_or_create since we definitely want the to rollback if there are any errors and can push a change to the branch if that's successful.